controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -117,6 +117,11 @@ void dynamic_channel_selection_task::work()
         break;
     }
     case eState::TRIGGER_SCAN: {
+        // get permission to scan
+        if (!database.try_lock_scan_permission(id)) {
+            break;
```

Comment:
nitpick, add an error message for logging

Reply:
this was intentionally not added since it will flood the controller log
I added a log of lock and release

a combination of those 2 with the state-change log print can provide all possible behavior states.